servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -710,11 +718,19 @@ public MergeResponse mergeRefIntoBranch(
 
       ResolvedHash fromRef =
           getHashResolver()
-              .resolveHashOnRef(fromRefName, fromHash, "Source hash", REQUIRED_UNAMBIGUOUS_HASH);
+              .resolveHashOnRef(
+                  fromRefName,
+                  fromHash,
+                  new HashValidator("Source hash").hashMustBePresent().hashMustNotBeAmbiguous());
```

Comment:
nit: maybe add a helper method like `nonAmbiguousHash(String fieldname)`?

Reply:
Now that `hashMustBePresent` has been merged with `hashMustNotBeAmbiguous` maybe that's not required anymore?